packer/builder/azure/driver_restapi/config.go

```diff
@@ -0,0 +1,200 @@
+	StorageAccountContainer string `mapstructure:"storage_account_container"`
+	OsType                  string `mapstructure:"os_type"`
+	OsImageLabel            string `mapstructure:"os_image_label"`
+	Location                string `mapstructure:"location"`
+	InstanceSize            string `mapstructure:"instance_size"`
+	UserImageLabel          string `mapstructure:"user_image_label"`
+
+	username         string `mapstructure:"username"`
+	tmpVmName        string
+	tmpServiceName   string
+	tmpContainerName string
+	userImageName    string
+}
+
+func newConfig(raws ...interface{}) (*Config, []string, error) {
+	var c Config
+
+	var md mapstructure.Metadata
+	err := config.Decode(&c, &config.DecodeOpts{
+		Metadata:    &md,
+		Interpolate: true,
+	}, raws...)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Defaults
+	log.Println(fmt.Sprintf("%s: %v", "PackerUserVars", c.PackerConfig.PackerUserVars))
+
+	if c.StorageAccountContainer == "" {
+		c.StorageAccountContainer = "vhds"
+	}
+
+	// Check values
+	var errs *packer.MultiError
+
+	if c.SubscriptionName == "" {
+		errs = packer.MultiErrorAppend(errs,
+			fmt.Errorf("subscription_name must be specified"))
+	}
+
+	if c.PublishSettingsPath == "" {
+		errs = packer.MultiErrorAppend(errs, fmt.Errorf("publish_settings_path must be specified"))
+	}
+
+	if c.StorageAccount == "" {
+		errs = packer.MultiErrorAppend(errs, fmt.Errorf("storage_account must be specified"))
+	}
+
+	if _, err := os.Stat(c.PublishSettingsPath); err != nil {
+		errs = packer.MultiErrorAppend(errs, fmt.Errorf("publish_settings_path is not a valid path: %s", err))
+	}
+
+	osTypeIsValid := false
+	osTypeArr := []string{
+		targets.Linux,
+		targets.Windows,
+	}
+
+	for _, osType := range osTypeArr {
+		if c.OsType == osType {
+			osTypeIsValid = true
+			break
+		}
+	}
+
+	if !osTypeIsValid {
+		errs = packer.MultiErrorAppend(errs,
+			fmt.Errorf("os_type is not valid, must be one of: %v", osTypeArr))
+	}
+
+	if c.OsImageLabel == "" {
+		errs = packer.MultiErrorAppend(errs, fmt.Errorf("os_image_label must be specified"))
+	}
+
+	if c.Location == "" {
+		errs = packer.MultiErrorAppend(errs, fmt.Errorf("location must be specified"))
+	}
+
+	sizeIsValid := false
+	instanceSizeArr := []string{
+		targets.ExtraSmall,
+		targets.Small,
+		targets.Medium,
+		targets.Large,
+		targets.ExtraLarge,
+		targets.A5,
+		targets.A6,
+		targets.A7,
+		targets.A8,
+		targets.A9,
+		targets.Standard_D1,
+		targets.Standard_D2,
+		targets.Standard_D3,
+		targets.Standard_D4,
+		targets.Standard_D11,
+		targets.Standard_D12,
+		targets.Standard_D13,
+		targets.Standard_D14,
+		targets.Standard_DS1,
+		targets.Standard_DS2,
+		targets.Standard_DS3,
+		targets.Standard_DS4,
+		targets.Standard_DS11,
+		targets.Standard_DS12,
+		targets.Standard_DS13,
+		targets.Standard_DS14,
+		targets.Standard_G1,
+		targets.Standard_G2,
+		targets.Standard_G3,
+		targets.Standard_G4,
+		targets.Standard_G5,
+	}
+
+	for _, instanceSize := range instanceSizeArr {
```

Comment:
in case when you find yourself in a position where you check multiple times if a `string` is in `[]string`,  you can write a method 

    func contains(s []string, v string) bool